src/index.js

```diff
@@ -11,26 +11,18 @@ import { assign } from './util';
  *    store.setState({ c: 'd' });   // logs { a: 'b', c: 'd' }
  */
 export default function createStore(state) {
-	let listeners = [];
+	const listeners = [];
 	state = state || {};
 
 	function unsubscribe(listener) {
-		let out = [];
-		for (let i=0; i<listeners.length; i++) {
-			if (listeners[i]===listener) {
-				listener = null;
-			}
-			else {
-				out.push(listeners[i]);
-			}
-		}
-		listeners = out;
+		const i = listeners.indexOf(listener);
+		if (i > -1) listeners.splice(i, 1);
```

Comment:
Simple array mutation

Reply:
This shifts the indices within the Array during event firing, which can cause the wrong set of listeners to be invoked.

Reply:
The listeners are called in descending order, so I think the shifted index is irrelevant - the other below it will still be called.